Make some of the Java classes proper utilities

vil02 opened the pull request:
## Purpose

Fixes: ballerina-platform/ballerina-standard-library#4921.

## Examples

## Checklist
- [x] Linked to an issue
- [ ] Updated the specification
- [x] Updated the changelog
- [ ] Added tests
- [ ] Checked native-image compatibility

vil02 commented:
@niveathika I have updated this branch. Could you please _approve_ the workflows again?

niveathika commented:
@vil02 since these classes are used in other DB connector, these changes must be migrated to ballerina distribution and verified before this can be merged.

vil02 commented:
> @vil02 since these classes are used in other DB connector, these changes must be migrated to ballerina distribution and verified before this can be merged.

@niveathika is there some action I need to take regarding that?

vil02 commented:
@niveathika @daneshk could you please remove the `stale` label?

vil02 commented:
> @niveathika @daneshk could you please remove the `stale` label?

Please `don't close`.

vil02 commented:
> > @niveathika @daneshk could you please remove the `stale` label?
> 
> Please `don't close`.

Also now.

vil02 commented:
😞

ThisaruGuruge commented:
> 😞
> 
> @niveathika is it possible to re-open this PR?

We are really very sorry for the late reply. I have reopened this.

vil02 commented:
> > > @niveathika @daneshk could you please remove the `stale` label?
> > 
> > 
> > Please `don't close`.
> 
> Also now.

Also now.

ThisaruGuruge commented:
> Also now.

Hi @vil02, it seems like the SonarCloud analysis has failed for your PR. Can you please check?

vil02 commented:
> Hi @vil02, it seems like the SonarCloud analysis has failed for your PR. Can you please check?

The complain from SonarCloud seems to a _false positive_. It complains that I have introduced a duplicated code. Sure... I have introduced `private` default constructors and other classes, like `QueryProcessor`, already have it. But it does not work like this.

In my opinion, this complain could be ignored. @ThisaruGuruge: what is you opinion?

daneshk commented:
> > Hi @vil02, it seems like the SonarCloud analysis has failed for your PR. Can you please check?
> 
> The complain from SonarCloud seems to a _false positive_. It complains that I have introduced a duplicated code. Sure... I have introduced `private` default constructors and other classes, like `QueryProcessor`, already have it. But it does not work like this.
> 
> In my opinion, this complain could be ignored. @ThisaruGuruge: what is you opinion?

Thanks for the contribution @vil02. SonarCloud complains about `QueryProcessor` which is not related to your fix. Let me check the usage of these classes in other db connectors and get back

daneshk commented on native/src/main/java/io/ballerina/stdlib/sql/utils/ProcedureCallResultUtils.java:
```diff
@@ -51,8 +51,8 @@
  */
-public class ProcedureCallResultUtils {
-    public ProcedureCallResultUtils() {
+public final class ProcedureCallResultUtils {
```
```suggestion
public class ProcedureCallResultUtils {
```

daneshk replied:
These two classes are extended in other SQL connector. Hence removing the final keyword

daneshk commented on native/src/main/java/io/ballerina/stdlib/sql/utils/RecordIteratorUtils.java:
```diff
@@ -36,8 +36,8 @@
  */
-public class RecordIteratorUtils {
-    public RecordIteratorUtils() {
+public final class RecordIteratorUtils {
```
```suggestion
public class RecordIteratorUtils {
```
